### PR TITLE
Portable layer changes

### PR DESCRIPTION
**Pull request by kar-rahul-aws**
Portable layer changes for Endpoint , as labs/ipV6 branch

Description
-----------
Ports changed:
1. DriverSAM
2. MPS2_AN385
3. ESP32
4. STM32
5. Zynq
6. Pic32mzef

Test Steps
-----------

Related Issue
-----------

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: source/portable/NetworkInterface/MPS2_AN385/NetworkInterface.c [hunk 1]
```diff
@@ -82,6 +82,22 @@ static void prvSetMACAddress( void );
 
 /*-----------------------------------------------------------*/
 
+/*
+ * A pointer to the network interface is needed later when receiving packets.
+ */
+static NetworkInterface_t * pxMyInterface;
+
+static BaseType_t xNetworkInterfaceInitialise( NetworkInterface_t * pxInterface );
```
Comment (htibosch) [inline source/portable/NetworkInterface/MPS2_AN385/NetworkInterface.c:90]: Would you be so kind to make these static function names unique in de source-tree by giving them some sort of a prefix?
Like for instance `xMPS2_NetworkInterfaceInitialise()`?
When every driver uses the same set of function names, a grep "xNetworkInterfaceInitialise" will yield more than 20 results.
And also, a name like `xMPS2_GetPhyLinkStatus()` shows immediately to which driver it belongs.

File: source/portable/NetworkInterface/STM32Hxx/NetworkInterface.c [hunk 2]
```diff
@@ -183,17 +181,16 @@ static size_t uxGetOwnCount( ETH_HandleTypeDef * heth );
...
-static BaseType_t xGetPhyLinkStatus( NetworkInterface_t * pxInterface );
+static BaseType_t xSTM32H_NetworkInterfaceOutput( NetworkInterface_t * pxInterface,
+                                                  NetworkBufferDescriptor_t * const pxBuffer,
+                                                  BaseType_t xReleaseAfterSend );
 
-NetworkInterface_t * pxFillInterfaceDescriptor( BaseType_t xEMACIndex,
-                                                NetworkInterface_t * pxInterface );
+static BaseType_t xSTM32H_GetPhyLinkStatus( NetworkInterface_t * pxInterface );
 
+NetworkInterface_t * pxSTM32Hxx_FillInterfaceDescriptor( BaseType_t xEMACIndex,
```
Comment (htibosch) [inline source/portable/NetworkInterface/STM32Hxx/NetworkInterface.c:192]: You're using two different prefixes here: `pxSTM32Hxx_FillInterfaceDescriptor()` and `xSTM32H_NetworkInterfaceOutput`.
I prefer the second one: `STM32H_`.

File: source/portable/NetworkInterface/WinPCap/NetworkInterface.c [hunk 3]
```diff
@@ -755,7 +755,7 @@ static BaseType_t xPacketBouncedBack( const uint8_t * pucBuffer )
 }
 /*-----------------------------------------------------------*/
 
-extern NetworkEndPoint_t* pxGetEndpoint( BaseType_t xIPType );
```
Comment (htibosch) [inline source/portable/NetworkInterface/WinPCap/NetworkInterface.c:758]: Strange that WinPCap/NetworkInterface.c shows many changes while it is equal to the dev/integration branch.

File: source/portable/NetworkInterface/esp32/NetworkInterface.c [hunk 4]
```diff
@@ -152,6 +220,8 @@ esp_err_t wlanif_input( void * netif,
     {
         /* Set the packet size, in case a larger buffer was returned. */
         pxNetworkBuffer->xDataLength = len;
+        pxNetworkBuffer->pxInterface = pxMyInterface;
+        pxNetworkBuffer->pxEndPoint = FreeRTOS_MatchingEndpoint( pxInterface, pcBuffer );
```
Comment (htibosch) [inline source/portable/NetworkInterface/esp32/NetworkInterface.c:224]: Here in 2 lines you're using `pxMyInterface` as well as `pxInterface`. It looks like only the former is available here?

**Review by htibosch — APPROVED** [review]
Thank you very much, Rahul !

The next time it would be easier for the reviewers to separate the uncrustify changes from the content changes. Now a reviewer has to check every "changed file" to see if the content has changed.
I am still preparing a revised of `FreeRTOS_MatchingEndpoint()`.

File: source/FreeRTOS_DNS_Cache.c [hunk 5]
```diff
@@ -561,58 +561,59 @@
     #endif /* ( ipconfigUSE_DNS_CACHE == 1 ) */
 /*-----------------------------------------------------------*/
 
-#if ( ipconfigDNS_CACHE_ADDRESSES_PER_ENTRY > 1 )
```
[the remark this answers is not in the document]
Comment (kar-rahul-aws) [reply source/FreeRTOS_DNS_Cache.c]: When I review the file with latest dev/integration FreeRTOS_DNS_Cache.c, I find no difference

File: source/portable/NetworkInterface/MPS2_AN385/NetworkInterface.c (hunk 1 above)
Comment (kar-rahul-aws) [reply source/portable/NetworkInterface/MPS2_AN385/NetworkInterface.c]: Done, thanks

File: source/portable/NetworkInterface/STM32Hxx/NetworkInterface.c (hunk 2 above)
Comment (kar-rahul-aws) [reply source/portable/NetworkInterface/STM32Hxx/NetworkInterface.c]: Done, thanks

File: source/portable/NetworkInterface/esp32/NetworkInterface.c (hunk 4 above)
Comment (kar-rahul-aws) [reply source/portable/NetworkInterface/esp32/NetworkInterface.c]: Done, thanks

File: source/portable/NetworkInterface/WinPCap/NetworkInterface.c (hunk 3 above)
Comment (kar-rahul-aws) [reply source/portable/NetworkInterface/WinPCap/NetworkInterface.c]: Made the fixes, currently in sync.

File: source/FreeRTOS_DNS_Cache.c (hunk 5 above)
[the remark this answers is not in the document]
Comment (kar-rahul-aws) [reply source/FreeRTOS_DNS_Cache.c]: Made the changes, fixed

**Review by shubnil — APPROVED** [review]

**Pull request merged**